### ISSUE TITLE
V parser segfault with `v -silent bug6.v`, discovered through radamsa

### ISSUE THREAD

**Issue by spytheman** [vlang/v#22093]
**V doctor:**
```
V full version: V 0.4.7 c92577e.75d65a0
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /space/v/oo
vexe: /space/v/oo/v
vexe mtime: 2024-08-21 10:37:35

vroot: OK, value: /space/v/oo
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.45.2
Git vroot status: weekly.2024.34-20-g423a94aa
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 0134e9b9

```

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg -silent bug6.v`
See attached file `bug6.v`

**What did you expect to see?**

V should not segfault

**What did you see instead?**
```
#1 13:16:35 ^ master /space/v/oo>./vdbg -silent bug6.v
Segmentation fault (core dumped)
```
[bug6.v.txt](https://github.com/user-attachments/files/16709071/bug6.v.txt)

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

**Comment by spytheman** [comment]
Still segfaults with latest V 18f89991efbeea490621a2a961d3b926f6592e8f .

**Comment by spytheman** [comment]
It is fixed by the latest V e3ceb5881a793c0924a4e3f03e6e2238906e7731 :
![image](https://github.com/user-attachments/assets/c198564a-e731-4f9a-acff-b748d711ecc5)
